### DISCUSSION TITLE
How to catch and log unexpected errors from the backend towards PostHog

### DISCUSSION THREAD

**Discussion by arik-dig** [digma-ai/digma-intellij-plugin#633] [Announcements]
Following issue #591 
I think we should have common way to response from our backend,
so the Plugin could read some structured data and report it to posthog.

**Possible solutions**

### Solution A

- on the backend side

1. At all our endpoints, on unexpected exceptions, it will respond with HTTP code of 555 ([the range of 500 are used for server codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#5xx_server_errors))
2. the body would contain some strucutred JSON which contains the exception and its stacktrace.

- on the plugin side

1. will check for HTTP code 555, and seeing such will read the structured body and report to PostHog

### Solution B

no one suggested yet

**Comment by borisermakof** [comment]
I believe we can end up with a problem when we send too much data to the posthog. Posthog is designed for the statistics, not logging. Maybe we should ask user for a permission to send error logs only from the backend to some centralised place(Coralogix or better hosted log system in our AWS)

**Comment by arik-dig** [comment]
@borisermakof maybe we can log the errors from the backend itself. 
the thing is that we want the details of the "plugin user".
so maybe we can add to all of our requests the context of the userId and log it when we log errors
